activities/GameOfLife.activity/js/activity.js

```diff
@@ -0,0 +1,191 @@
+  document.querySelector('.generation-status').innerText = l10n.get('Generation');
+
+  board.draw();
+
+  var storeLocally = function storeLocally(state) {
+    dataStore.setDataAsText({
+      state: state
+    });
+    console.log('writing');
+    dataStore.save(function (err) {
+      if (err) {
+        console.log('writing failed.');
+        console.error(err);
+      } else {
+        console.log('writing saved.');
+      }
+    });
+  };
+
+  var generateGeneration = function generateGeneration() {
+    if (state.state.shouldPlay) {
+      var nextGenerationBoard = state.state.boardState.map(function (row, y) {
+        return row.map(function (cell, x) {
+          return transformCellByRule(cell, findNeighbours(x, y));
+        });
+      });
+      state.set(function (prev) {
+        return {
+          boardState: nextGenerationBoard,
+          generation: prev.generation + 1
+        };
+      });
+      setTimeout(generateGeneration, 100);
+    } else {
+      return 0;
+    }
+  };
+
+  var transformCellByRule = function transformCellByRule(cell, neighbours) {
+    var cellIsAlive = cell === 1 || cell === 2;
+    var aliveInNeighbour = neighbours.filter(function (cell) {
+      return cell === 1 || cell === 2;
+    });
+    var numOfAliveNeighbours = aliveInNeighbour.length;
+    if (cellIsAlive) {
+      if (numOfAliveNeighbours < 2) {
+        return 0;
+      } else if (numOfAliveNeighbours === 2 || numOfAliveNeighbours === 3) {
+        return 1;
+      } else {
+        return 0;
+      }
+    } else {
+      if (numOfAliveNeighbours === 3) {
+        return 2;
+      } else {
+        return 0;
+      }
+    }
+  };
+
+  var findNeighbours = function findNeighbours(x, y) {
+    var leftX = x - 1 === -1 ? 49 : x - 1;
+    var rightX = x + 1 === 50 ? 0 : x + 1;
+    var topY = y - 1 === -1 ? 29 : y - 1;
+    var bottomY = y + 1 === 30 ? 0 : y + 1;
+    var boardState = state.state.boardState;
+
+    var left = boardState[y][leftX];
+    var right = boardState[y][rightX];
+    var top = boardState[topY][x];
+    var bottom = boardState[bottomY][x];
+    var leftTop = boardState[topY][leftX];
+    var leftBottom = boardState[bottomY][leftX];
+    var rightTop = boardState[topY][rightX];
+    var rightBottom = boardState[bottomY][rightX];
+
+    return [left, right, top, bottom, leftTop, leftBottom, rightTop, rightBottom];
+  };
+
+  state.subscribe({
+    boardState: ['.fake-selector', function (fakeElem, value, prevValue) {
+      board.update(value);
+    }],
+
+    generation: ['.generation-count', 'innerText'],
+
+    playPauseIcon: ['#play-pause', function (elem, value, prevValue) {
+      elem.className = value + ' toolbutton';
+    }],
+
+    shouldPlay: ['.fake-selector', function (fakeElem, value, prevValue) {
+      if (value) {
+        generateGeneration();
+      }
+    }]
+  });
+
+  dataStore.loadAsText(function (err, metadata, data) {
+    var dataToLoad = data ? data.state : randomPattern();
```

Comment:
it seems that state is in data.state.state so need to be replaced by:
`    var dataToLoad = data && data.state ? data.state.state : randomPattern();`